packages/contracts-bedrock/scripts/ChainAssertions.sol

```diff
@@ -163,25 +163,42 @@ library ChainAssertions {
     }
 
     /// @notice Asserts the OptimismPortal is setup correctly
-    function checkOptimismPortal(Types.ContractSet memory _contracts, DeployConfig _cfg, bool _isProxy) internal view {
+    function checkOptimismPortal(
+        Types.ContractSet memory _contracts,
+        DeployConfig _cfg,
+        bool _isProxy,
+        bool _isInitialized
+    )
+        internal
+        view
+    {
         console.log("Running chain assertions on the OptimismPortal");
+
         OptimismPortal portal = OptimismPortal(payable(_contracts.OptimismPortal));
 
         address guardian = _cfg.superchainConfigGuardian();
         if (guardian.code.length == 0) {
             console.log("Guardian has no code: %s", guardian);
         }
 
```

Comment:
I am confused as to why we need `_isInitialized`

Reply:
`checkOptimismMintableERC20Factory` is called at deployment of OptimismMintableERC20Factory https://github.com/ethereum-optimism/optimism/blob/9b513d7c18048bceeb132c1796cf1975fd332e91/packages/contracts-bedrock/scripts/Deploy.s.sol#L555 however, since at deployment the storage variables are set to null values now, checkOptimismMintableERC20Factory fails because the addresses don't match with dependent contracts. there are two options: removing the call to checkOptimismMintableERC20Factory from deployment i.e., removing this line: https://github.com/ethereum-optimism/optimism/blob/9b513d7c18048bceeb132c1796cf1975fd332e91/packages/contracts-bedrock/scripts/Deploy.s.sol#L555 or making checkOptimismMintableERC20Factory aware of the fact that OptimismMintableERC20Factory isn't initialized. originally, I had gone for the first approach, but this comment made me change my mind (highly recommend reading for context): https://github.com/ethereum-optimism/optimism/pull/8631#discussion_r1428267234

Reply:
`checkOptimismPortal` is called at deployment of OptimismPortal https://github.com/ethereum-optimism/optimism/blob/bb5e92e5ee49aa5099c463a391eeb33bd6308d6f/packages/contracts-bedrock/scripts/Deploy.s.sol#L503 however, since at deployment the storage variables are set to null values now, checkOptimismPortal fails because the addresses don't match with dependent contracts. there are two options: removing the call to checkOptimismPortal from deployment i.e., removing this line: https://github.com/ethereum-optimism/optimism/blob/bb5e92e5ee49aa5099c463a391eeb33bd6308d6f/packages/contracts-bedrock/scripts/Deploy.s.sol#L503 or making `checkOptimismPortal` aware of the fact that OptimismPortal isn't initialized. originally, I had gone for the first approach, but this comment made me change my mind (highly recommend reading for context): https://github.com/ethereum-optimism/optimism/pull/8631#discussion_r1428267234. thoughts?

Reply:
realized you were right 🙂